google/cloud/bigtable/results.h

```diff
@@ -54,6 +55,39 @@ class ResultSourceInterface {
   Metadata() = 0;
 };
 
+/**
+ * Represents the stream of `Rows` returned from `bigtable::Table::ReadRows()`.
+ *
+ * This is a range defined by the [Input Iterators][input-iterator] returned
+ * from its `begin()` and `end()` members. Callers may directly iterate the
+ * `RowStream` instance, which will return a sequence of `StatusOr<QueryRow>`
+ * objects.
+ *
+ * [input-iterator]: https://en.cppreference.com/w/cpp/named_req/InputIterator
+ */
+class RowStream {
+ public:
+  RowStream() = default;
+  explicit RowStream(std::unique_ptr<ResultSourceInterface> source)
+      : source_(std::move(source)) {}
+
+  // This class is movable but not copyable.
+  RowStream(RowStream&&) = default;
+  RowStream& operator=(RowStream&&) = default;
+
+  /// Returns a `QueryRowStreamIterator` defining the beginning of this range.
+  QueryRowStreamIterator begin() {
```

Comment:
Let's keep this as RowStreamIterator. I'm hoping we can eventually factor out RowStream and RowStreamIterator to a common library that both Bigtable and Spanner can use.